app/feature/travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/AddCoInsured.kt

```diff
@@ -18,26 +22,23 @@ import com.hedvig.android.core.designsystem.component.textfield.HedvigTextField
 import com.hedvig.android.core.designsystem.material3.squircle
 import com.hedvig.android.core.designsystem.preview.HedvigPreview
 import com.hedvig.android.core.designsystem.theme.HedvigTheme
-import com.hedvig.android.core.ui.ValidatedInput
 import com.hedvig.android.core.ui.clearFocusOnTap
 import com.hedvig.android.core.ui.scaffold.HedvigScaffold
-import com.hedvig.android.feature.travelcertificate.TravelCertificateUiState
-import com.hedvig.android.feature.travelcertificate.data.TravelCertificateResult
+import com.hedvig.android.feature.travelcertificate.CoInsured
 import hedvig.resources.R
-import kotlinx.datetime.LocalDate
+import java.util.*
 
 @Composable
 fun AddCoInsured(
-  selectedCoInsuredId: String?,
-  uiState: TravelCertificateUiState,
+  coInsured: CoInsured?,
   navigateBack: () -> Unit,
-  onSsnChanged: (String) -> Unit,
-  onNameChanged: (String) -> Unit,
-  onRemoveClicked: (String) -> Unit,
-  onSaveClicked: () -> Unit,
+  onRemoveCoInsured: (String) -> Unit,
+  onEditCoInsured: (CoInsured) -> Unit,
+  onAddCoInsured: (CoInsured) -> Unit,
 ) {
 
-  val selectedCoInsured = uiState.coInsured.input.firstOrNull { it.id == selectedCoInsuredId }
+  var name by remember { mutableStateOf(coInsured?.name ?: "") }
+  var ssn by remember { mutableStateOf(coInsured?.ssn ?: "") }
```

Comment:
Since we're keeping this state here inside the composable itself and not the VM, it's a good idea to do a `rememberSaveable` instead of `remember` so it still survives configuration changes like it would if it were in the VM.